src/controller/migrate/materialization/mod.rs

```diff
@@ -404,6 +404,21 @@ impl Materializations {
         assert!(replay_obligations.is_empty());
     }
 
+    /// Retrieves the materialization status of a given node, or None
+    /// if the node isn't materialized.
+    pub fn get_status(&self, index: &NodeIndex, node: &Node) -> MaterializationStatus {
+        let is_materialized = self.have.contains_key(index)
+            || node.with_reader(|r| r.is_materialized()).unwrap_or(false);
```

Comment:
Correct, but aren't readers always materialized? (`is_materialized`returns a bool, so it doesn't differentiate full/partial either)